src/main/java/com/linkedin/kafka/clients/largemessage/MessageSplitterImpl.java

```diff
@@ -78,7 +78,7 @@ public MessageSplitterImpl(int maxSegmentSize,
     int messageSizeInBytes = serializedRecord.length;
     ByteBuffer bytebuffer = ByteBuffer.wrap(serializedRecord);
 
-    byte[] segmentKey = key == null ? LiKafkaClientsUtils.uuidToBytes(segmentMessageId) : key;
+    byte[] segmentKey = (key == null && numberOfSegments > 1) ? LiKafkaClientsUtils.uuidToBytes(segmentMessageId) : key;
```

Comment:
Right, this is to preserve the null key behavior so that it won't be accidentally set to a UUID.